com.unity.render-pipelines.high-definition/Runtime/RenderPipelineResources/HDUserSettings.cs

```diff
@@ -71,8 +80,11 @@ public static bool wizardNeedRestartAfterChangingToDX12
 
         static HDUserSettings CreateOrLoad()
         {
-            // force loading of HDProjectSetting first: its migration can create and init this file
-            int unused = HDProjectSettings.materialVersionForUpgrade;
+            // Note: HDProjectSettings should load first but it sits in an editor assembly
+            // but its instance creation is triggered early on DLL load and this is sufficient.
+            // Otherwise we could check if the HDProjectSettingsProxy.projectSettingsFolderPath delegate is null,
+            // and force the assembly load otherwise.
+            // HDProjectSettings' migration can create and init the HDUserSettings file.
```

Comment:
@sebastienlagarde See HDProjectSettings.cs returned to editor assembly and the InitializeOnLoadMethod method.
@RSlysz for visibility.

Reply:
Also note: I guess this will work as long as HDUserSettings clients are not themselves using [InitializeOnLoadMethod] in one of their code path and Unity happens to enumerate the method with the attribute in the runtime assembly before the editor assembly. If that ever happens, I guess we can test it here with a null test on the proxy function pointer, and a clumsy Assembly.Load here in that case.